scripts/us_epa/ghgrp/download.py

```diff
@@ -0,0 +1,140 @@
+            self._extract_data(headers)
+        for sheet, csv_name in SHEET_NAMES_TO_CSV_FILENAMES.items():
+            headers_df = pd.DataFrame.from_dict(headers[sheet], orient='index')
+            headers_df.transpose().to_csv(os.path.join(SAVE_PATH, f'cols_{csv_name}'), index=None)
+
+    def save_all_crosswalks(self):
+        """Builds individual year crosswalks, as well as a join crosswalk for all years."""
+        print(f'Saving all ID crosswalks')
+        crosswalks = []
+        for current_year in self.years:
+            crosswalks.append(downloader._save_crosswalk())
+        all_crosswalks_df = pd.concat(crosswalks, join='outer')
+        all_crosswalks_df = all_crosswalks_df.sort_values(
+            by=[GHGRP_ID_COL, 'FRS Id', 'ORIS CODE'])
+        all_crosswalks_df = all_crosswalks_df.drop_duplicates()
+        all_crosswalks_df.to_csv(os.path.join(SAVE_PATH, 'all_crosswalks.csv'),
+                                 header=True,
+                                 index=None)
+        return all_crosswalks_df
+
+    def _csv_path(self, csv_filename):
+        return os.path.join(SAVE_PATH, f'{self.current_year}_{csv_filename}')
+
+    def _extract_data(self, headers):
+        summary_filename = os.path.join(
+            SAVE_PATH, YEAR_DATA_FILENAME.format(year=self.current_year))
+        xl = pd.ExcelFile(summary_filename)
+        for sheet in xl.sheet_names:
+            csv_filename = SHEET_NAMES_TO_CSV_FILENAMES.get(sheet, None)
+            if not csv_filename:
+                print(f'Skipping sheet: {sheet}')
+                continue
+            summary_file = xl.parse(sheet, header=HEADER_ROW, dtype=str)
+            summary_file.to_csv(self._csv_path(csv_filename),
+                                index=None,
+                                header=True)
+            headers[sheet][self.current_year] = summary_file.columns
+
+    def _save_crosswalk(self):
+        oris_df = pd.read_excel(CROSSWALK_URI,
+                                'ORIS Crosswalk',
+                                header=0,
+                                dtype=str,
+                                usecols=CROSSWALK_COLS_TO_KEEP)
+        oris_df = oris_df.rename(columns={'GHGRP Facility ID': GHGRP_ID_COL})
+        all_facilities_df = pd.DataFrame()
+        for sheet, csv_filename in SHEET_NAMES_TO_CSV_FILENAMES.items():
+            csv_path = self._csv_path(csv_filename)
+            if not os.path.exists(csv_path):
+                continue
+            df = pd.read_csv(csv_path,
+                             usecols=[GHGRP_ID_COL, 'FRS Id'],
+                             dtype=str)
+            all_facilities_df = all_facilities_df.append(df)
+        all_facilities_df = all_facilities_df.join(
+            oris_df.set_index(GHGRP_ID_COL), on=GHGRP_ID_COL, how='left')
+        all_facilities_df.to_csv(self._csv_path('crosswalk.csv'),
+                                 header=True,
+                                 index=None)
```

Comment:
do we need this to_csv ?

Reply:
ah, not anymore. initially i was thinking of processing each year horizontally, but it makes more sense to do it by sheet across years. deleted.